src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -499,7 +520,17 @@ def makeTransferRucio(self, wflow, dataIn, dids, dataSize, grouping, copies, nod
             self.logger.info("Creating rule for workflow %s with %d DIDs in container %s, RSEs: %s, grouping: %s",
                              wflow.getName(), len(dids), dataIn['name'], rseExpr, grouping)
             try:
-                res = self.rucio.createReplicationRule(dids, rseExpr, **ruleAttrs)
+                # make decision about current workflow, if it is new request we'll create
+                # new replication rule, otherwise we'll move replication rule
+                if wflow.updatedWorkflow:
+                    rules = self.rucio.listDataRules(wflow.getName())
```

Comment:
This line is in practice asking Rucio for rules for a given workflow, which won't work.

Instead, I suggest you to send a GET request to retrieve the transfer document (created and stored by WM) from MSTransferor. Example API and data structure is:
https://cmsweb.cern.ch/ms-transferor/data/info?request=cmsunified_HARVEST-Run2022E_ZeroBias_30May2023v2_240123_140316_7935

Hence, you need to consume `transferDoc` -> `transfers` -> `transferIDs`, keeping in mind that there could be more than one element under the `transfers` field (even though in practice I think it will be a single one).

Reply:
ok, added new method to fetch transfer IDs.